src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamSungSmartThingsAggregatorCommunicator.java

```diff
@@ -0,0 +1,1696 @@
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.common.location.LocationManagementMetric;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.common.room.CreateRoomMetric;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.common.room.RoomManagementMetric;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.Hub;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.device.Device;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.device.DeviceHealth;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.device.DeviceWrapper;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.location.Location;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.location.LocationWrapper;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.presentation.DashboardPresentation;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.presentation.DetailViewPresentation;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.presentation.DevicePresentation;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.presentation.Language;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.presentation.PoCode;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.room.Room;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.room.RoomWrapper;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.sence.Scene;
+import com.avispl.symphony.dal.infrastructure.management.samsung.smartthings.dto.sence.SceneWrapper;
+import com.avispl.symphony.dal.util.StringUtils;
+
+/**
+ * SamSungSmartThingsAggregatorCommunicator
+ *
+ * @author Kevin / Symphony Dev Team<br>
+ * Created on 7/22/2022
+ * @since 1.0.0
+ */
+public class SamSungSmartThingsAggregatorCommunicator extends RestCommunicator implements Aggregator, Monitorable, Controller {
+
+	/**
+	 * Process that is running constantly and triggers collecting data from Zoom API endpoints, based on the given timeouts and thresholds.
+	 *
+	 * @author Maksym.Rossiytsev
+	 * @since 1.0.0
+	 */
+	class SamSungSmartThingsDeviceDataLoader implements Runnable {
+		private volatile boolean inProgress;
+		private volatile int threadIndex;
+		private List<String> deviceIds;
+
+		/**
+		 * Parameters constructors
+		 *
+		 * @param threadIndex index of thread
+		 * @param deviceIds list of device IDs
+		 */
+		public SamSungSmartThingsDeviceDataLoader(int threadIndex, List<String> deviceIds) {
+			inProgress = true;
+			this.threadIndex = threadIndex;
+			this.deviceIds = deviceIds;
+		}
+
+		@Override
+		public void run() {
+			mainloop:
+			while (inProgress) {
```

Comment:
There is no need to do a while loop here, because we already submit 8 threads from the beginning (getMultiple)

Reply:
I fixed it